File: basics/roscpp/simple_pub/src/simple_pub_class.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

class Publisher
{
public:

	Publisher()
	{
		ROS_INFO("%s\n", "I'm The Publisher Object");

		pub_ = nh_.advertise<std_msgs::String>("/talk", 1);
	}

	void
	doWork()
	{
		std_msgs::String msg;
		msg.data = "Hello World";

		pub_.publish(msg);
	}

private:
	ros::NodeHandle nh_;
	ros::Publisher pub_;
};

int
main(int argc, char **argv)
{
	ros::init(argc, argv, "simple_pub_class_node");

	Publisher pub;

	//2Hz rate

	ros::Rate loop_rate(2);

	while(ros::ok()){
		pub.doWork();
		ros::spinOnce();
		loop_rate.sleep();
	}

	exit(EXIT_SUCCESS);
}

File: basics/rospy/simple_pub_py/src/simple_pub_class.py
#! /usr/bin/env python

import rospy
from std_msgs.msg import String

class Publisher():
    def __init__(self):
        self.pub = rospy.Publisher("/talk", String, queue_size=1)

    def doWork(self):
        msg = String()
        msg.data = "Hello World"
        self.pub.publish(msg)

def main():
    rospy.init_node("simple_pub_class_node")

    publisher = Publisher()

    # 2Hz Rate:

    rate = rospy.Rate(2)
    while(not rospy.is_shutdown()):
        publisher.doWork()
        rate.sleep()

if __name__ == "__main__":

    try:
        main()
    except rospy.ROSInterruptException:
        pass
